sdk/logs/src/main/java/io/opentelemetry/sdk/logs/ReadWriteLogRecord.java

```diff
@@ -47,7 +53,87 @@ default ReadWriteLogRecord setAllAttributes(Attributes attributes) {
   /** Return an immutable {@link LogRecordData} instance representing this log record. */
   LogRecordData toLogRecordData();
 
-  // TODO: add additional log record accessors. Currently, all fields can be accessed indirectly via
-  // #toLogRecordData() at the expense of additional allocations.
+  /**
+   * Returns the value of a given attribute if it exists. This is the equivalent of calling
+   * getAttributes().get(key)
+   */
+  @Nullable
+  default <T> T getAttribute(AttributeKey<T> key) {
+    return toLogRecordData().getAttributes().get(key);
+  }
+
+  /** Returns the resource of this log record. */
+  @Override
+  default Resource getResource() {
+    return toLogRecordData().getResource();
+  }
+
+  /** Returns the instrumentation scope that generated this log. */
+  @Override
+  default InstrumentationScopeInfo getInstrumentationScopeInfo() {
+    return toLogRecordData().getInstrumentationScopeInfo();
+  }
 
+  /** Returns the timestamp at which the log record occurred, in epoch nanos. */
+  @Override
+  default long getTimestampEpochNanos() {
+    return toLogRecordData().getTimestampEpochNanos();
+  }
+
+  /** Returns the timestamp at which the log record was observed, in epoch nanos. */
+  @Override
+  default long getObservedTimestampEpochNanos() {
+    return toLogRecordData().getTimestampEpochNanos();
+  }
+
+  /** Return the span context for this log, or {@link SpanContext#getInvalid()} if unset. */
+  @Override
+  default SpanContext getSpanContext() {
+    return toLogRecordData().getSpanContext();
+  }
+
+  /** Returns the severity for this log, or {@link Severity#UNDEFINED_SEVERITY_NUMBER} if unset. */
+  @Override
+  default Severity getSeverity() {
+    return toLogRecordData().getSeverity();
+  }
+
+  /** Returns the severity text for this log, or null if unset. */
+  @Nullable
+  @Override
+  default String getSeverityText() {
+    return toLogRecordData().getSeverityText();
+  }
+
+  @Deprecated
+  @Override
+  @SuppressWarnings("deprecation") // Implementation of deprecated method
+  default io.opentelemetry.sdk.logs.data.Body getBody() {
+    return toLogRecordData().getBody();
+  }
+
+  /** Returns the {@link Value} representation of the log body, of null if unset. */
+  @Nullable
+  @Override
+  default Value<?> getBodyValue() {
+    return toLogRecordData().getBodyValue();
+  }
+
+  /** Returns the attributes for this log, or {@link Attributes#empty()} if unset. */
+  @Override
+  default Attributes getAttributes() {
```

Comment:
`ReadableSpan` also has a method for accessing a single attribute, which would be good to add here. 
```
@Nullable
<T> T getAttribute(AttributeKey<T> key);
```

Reply:
Yes, that exists above. 👍🏻